src/main/java/net/minecraftforge/fml/client/SplashProgress.java

```diff
@@ -831,4 +841,48 @@ else if(fmlPack.resourceExists(loc))
         }
         return mcPack.getInputStream(loc);
     }
+
+    private static void drawMemoryWarning()
+    {
+        long maxMemory = Runtime.getRuntime().maxMemory();
+        long totalMemory = Runtime.getRuntime().totalMemory();
+        long freeMemory = Runtime.getRuntime().freeMemory();
+        long usedMemory = totalMemory - freeMemory;
+        long usedMemoryPercent = usedMemory * 100L / maxMemory;
+        if (usedMemoryPercent >= memWarnPercent)
+        {
+            showMemoryWarning = true;
+        }
+
+        if (showMemoryWarning)
+        {
+            glPushMatrix();
+            glScalef(2, 2, 1);
+
+            List<String> list = Lists.newArrayList(
```

Comment:
I'm not sure it matters really, your game is already most likely going to end up running horribly when this message comes up because you're running out of memory.